lib/two_factor_authentication/models/two_factor_authenticatable.rb

```diff
@@ -73,8 +74,14 @@ def totp_enabled?
           respond_to?(:otp_secret_key) && !otp_secret_key.nil?
         end
 
+        def confirm_totp_secret(secret, code, options = {})
+          return false unless authenticate_totp(code, {otp_secret_key: secret})
+          self.otp_secret_key = secret
+          true
+        end
+
         def generate_totp_secret
-          self.otp_secret_key = ROTP::Base32.random_base32
+          ROTP::Base32.random_base32
         end
```

Comment:
This method is referenced in the README.   I think the flow could use more documentation though.

I prefer the idea of storing the unconfirmed secret in the session.. there is not need to persist it outside the lifetime of the session (unlike unconfirmed email that devise uses for example).